compose/ui/ui-graphics/src/skikoMain/kotlin/androidx/compose/ui/graphics/layer/SkiaGraphicsLayer.skiko.kt

```diff
@@ -332,6 +333,7 @@ actual class GraphicsLayer internal constructor(
     }
 
     private fun recordWithTracking(block: (SkiaBackedCanvas) -> Unit) {
+        val renderNode = renderNode ?: return
```

Comment:
We can hide the actual error if we allow calling `record` after the layer is released.

Fail-fast approach usually helps more than allowing calling methods on an invalid-stated object. Calling `record` that actually doesn't record means that the state is invalid.

Do you know why `GraphicsLayerOwnerLayer.record` is called after `GraphicsLayerOwnerLayer.release`? Can the error be in `commonMain`?

Reply:
There are a few things here
1. Public API for manual management/releasing
2. Valid cases where it was released by user but still used in hierarchy (see that real disposal of objects happens not in release method)

So we should definitely add guards here and avoid throwing error in case of usages after release.

My guess it something about order of callbacks in the last frame or something. So yes, it might be some issue in common code, but I believe it should behave like it does after this PR even in this case.

Anyway we need to fix a crash here, it won't hide anything really serious

Reply:
Looked at the stack trace again - it's not record after release. It's multiple releasing of child layer...

Maybe something like:
```
parentLayerUsages-- 2
parentLayerUsages-- 1
parentLayerUsages-- 0 // Release
parentLayerUsages++ 1
parentLayerUsages-- 0 // Release
```

This doesn't feel super robust, but in the case of refactoring, it's better to do it in sync with AOSP/Android

Reply:
How Android behave in this situation?

```
    val graphicsContext = LocalGraphicsContext.current
    LaunchedEffect(Unit) {
        val layer = graphicsContext.createGraphicsLayer()
        graphicsContext.releaseGraphicsLayer(layer)
        layer.record...
    }
```
Just reading the docs, it says:
<img width="509" alt="image" src="https://github.com/user-attachments/assets/11ba8c5b-d582-4239-b7f9-53f4bda082c2" />

But does it throw an exception? If not, I think it is not good, but then we also have to not throw it to preserve the same behaviour and potential errors in `commonMain`